Rakefile

```diff
@@ -300,6 +300,30 @@ begin
     puts 'All the gems are up to date' unless has_pending_releases
   end
 
+  # Task topological_order
+  #-----------------------------------------------------------------------------#
+
+  desc 'Print the topological ordering of all the Rainforest gems'
+  task :topological_order do
+    gems = Hash[gem_dirs.map { |repo| [gem_name(repo), repo] }]
+    gemfile = proc do
+      gems.each do |name, repo|
+        gem name, :path => repo
+      end
+    end
+    require 'bundler'
+    old_root = Bundler.method(:root)
+    def Bundler.root
+      Bundler::SharedHelpers.pwd.expand_path
+    end
+    builder = Bundler::Dsl.new
+    builder.instance_eval(&gemfile)
+    definition = builder.to_definition(nil, true)
+    puts definition.specs.map { |s| gems[s.name] }.compact
+    bundler_module = class << Bundler; self; end
+    bundler_module.send(:define_method, :root, old_root)
```

Comment:
Might be nice to use some whitespace and break logical chunks to make this 17 continuous line code-block more readable.